### DISCUSSION TITLE
Maya Standard Surface Shader

### DISCUSSION THREAD

**Discussion by wileyknight** [iimachines/Maya2glTF#156] [Q&A]
There seems to be a problem exporting standard surface shaders. I'm getting blinn, phong, and lambert shaders just fine.

aiStandard surface aren't exporting though I wasn't expecting them to since they are Arnold shaders. However, I was expecting to see the basic Standard Surface shader.

**Comment by ziriax** [comment]
No that "Uber shader" (as Autodesk calls it) is not supported yet. I has many features that glTF 2.0 doesn't support.

You can use our  glTF2.0 compatible hardware shader though (OpenGL only, not Direct3D).

Maybe you have a scene that requires the standard shader and want to export to glTF too?

**Reply by wileyknight** [reply]
I have a scene that I am exporting for three.js and was thinking that the basic (non arnold) standard surface shader would be compatible with the three.js meshStandardMaterial shader as both of them work from the same base of Color, Roughness, Metalness. There are no other comparable shader overlaps. I figured the options from maya that are not supported would simply be dropped as this is a "standard" shader.

I can use phong and blinn if I have to or even just not use shaders from glTF but it would be nice if everything was in the file.

**Comment by ziriax** [comment]
Yes, that makes sense.

It shouldn’t be too hard to support this.

PRs are welcome, otherwise I will take a look at this myself

**Reply by wileyknight** [reply]
I don't know enough about converting the shaders to know where to start on this. Do you think you'll be able to find some time to work on this?

**Reply by ziriax** [reply]
It depends on what properties from the shader you would like too support.

I don't know the shader myself, so I will have to learn it too 😉
